Src/FluentAssertions/Execution/TUnitFramework.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace FluentAssertions.Execution;
+
+internal class TUnitFramework : LoadableTestFramework
+{
+    protected override string ExceptionFullName => "TUnit.Assertions.Exceptions.AssertionException";
+
+    protected internal override string AssemblyName => "TUnit.Assertions";
```

Comment:
Shouldn't you also make sure that the `TUnit.Core` or `TUnit.Engine` assembly is loaded, as you mentioned [here](https://github.com/thomhurst/TUnit/issues/580#issuecomment-2406669665) that it is possible to use the assertions library with a different test framework.
So if I use the TUnit assertions within my xunit test project, it should still throw the xunit exceptions, right?

Reply:
Well, wouldn't you want to throw relevant exceptions to the assertion library you're using?

Reply:
Consider I use xunit (so have `[Fact]` attributes), but include the `TUnit.Assertions` library, because I like the assertions, so I would have a test like the following:
```csharp
// xunit test runner
[Fact]
public async Task DummyTest()
{
  // TUnit assertions library
  await Assert.That(true).IsFalse();
}
```
As the test runner will be xunit, it should throw a `XunitException`, but with the current implementation it would throw the exception from TUnit which xunit knows nothing about.

Reply:
Hmm... I don't expect this to be a very common scenario. The user can always force the fluent assertion framework can't they?

https://fluentassertions.com/introduction#detecting-test-frameworks

Reply:
Why should this be a niche scenario? I really like the "await always approach" from TUnit and could see me using it in project without switching to the TUnit engine (maybe because I need to use the old vstest runner)...

And it should be quite easy to detect that the `TUnit.Engine` class is loaded and if not to not use the `TUnit` exceptions.